### PR TITLE
Allow qualified exports to extended modules

### PR DESCRIPTION
**Pull request by rjernst**
In #94884 the ability to add qualified exports and opens from jars upstream of server was added. Some Elasticsearch components need to qualify their exports to another component. This commit tweaks the exports handling in PluginsService to also handle plugins with exports/opens.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/module-info.java [hunk 1]
```diff
@@ -213,5 +213,6 @@
     exports org.elasticsearch.xpack.core.watcher.watch;
     exports org.elasticsearch.xpack.core.watcher;
 
-    opens org.elasticsearch.license.internal; // spi
+    // opens org.elasticsearch.license.internal to org.elasticsearch.server; // spi
```
Comment (jakelandis) [inline x-pack/plugin/core/src/main/java/module-info.java:216]: nit: delete the commented out opens

**Review by jakelandis — COMMENTED** [review]
I think I understand the change here (LGTM), but probably not enough so to approve the PR.  

Thanks again for fixing this !  I tested locally and and works great.

**Comment by rjernst** [comment]
@elasticsearchmachine run elasticsearch-ci/part-2

**Comment by ChrisHegarty** [comment]
I think this PR is fine as is, but there is an alternative...

In 94884, it was necessary for the libs to provide some way of modifying their exports/opens available - this was done explicitly through a service. Since the libs module is loaded in the boot layer (along with the jdk modules, server, etc), it is not possible for server to modify it's module exports directly - there is no access to a module layer controller for the boot layer.

For qualified exports/opens between ES plugin modules, then we could use the module layer controller to reify the qualified exports automagically, without the need for each plugin module to implement the exports service. We can do this since it is our server that creates the module layers for the plugin modules (where as for libs, it is the JDK that creates the boot layer).

**Comment by rjernst** [comment]
Thanks Chris! I applied your idea in [8b44d57](https://github.com/elastic/elasticsearch/pull/95243/commits/8b44d57ecee9f41b31165b2d12c01c0f81bb0008), let me know what you think.

**Review by ChrisHegarty — APPROVED** [review]
I think that this looks good. And is a great improvement as qualified exports/opens between ES modules not in the boot layer will just work automatically. 👍

**Pull request merged**